rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/History.scala

```diff
@@ -58,9 +82,32 @@ final case class PointerBlock private (toVector: Vector[TriePointer]) extends No
 
   def countNonEmpty: Int = toVector.count(_ != EmptyPointer)
 
-  def hash: Blake2b256Hash = Blake2b256Hash.create(encode.toByteVector)
+  @SuppressWarnings(Array("org.wartremover.warts.Var"))
+  @transient
+  private[this] var _data: BitVector = null
 
-  def encode: BitVector = PointerBlock.codecPointerBlock.encode(this).get
+  @SuppressWarnings(Array("org.wartremover.warts.Var"))
+  @transient
+  private[this] var _hash: Blake2b256Hash = null
+
+  private[this] def calculateHash() = {
+    _data = PointerBlock.codecPointerBlock.encode(this).get
+    _hash = Blake2b256Hash.create(_data.toByteVector)
+  }
+
+  def hash: Blake2b256Hash = {
+    if (_data == null) {
```

Comment:
`if(_hash == null)`

Reply:
hash depends on data being calculated. if data is there we calculate both data and hash.